embedded/document/engine.go

```diff
@@ -938,3 +938,79 @@ func (e *Engine) readMetadataAndValue(key []byte, atTx uint64, skipIntegrityChec
 
 	return entry.Metadata(), v, nil
 }
+
+func (e *Engine) DeleteDocument(ctx context.Context, collectionName string, query *protomodel.Query) (err error) {
+	sqlTx, err := e.sqlEngine.NewTx(ctx, sql.DefaultTxOptions())
+	if err != nil {
+		return mayTranslateError(err)
+	}
+	defer sqlTx.Cancel()
+
+	table, err := getTableForCollection(sqlTx, collectionName)
+	if err != nil {
+		return mayTranslateError(err)
+	}
+
+	idFieldName := docIDFieldName(table)
+
+	queryCondition, err := e.generateSQLExpression(ctx, query, table)
+	if err != nil {
+		return mayTranslateError(err)
+	}
+
+	queryStmt := sql.NewSelectStmt(
+		[]sql.Selector{sql.NewColSelector(collectionName, idFieldName)},
+		collectionName,
+		queryCondition,
+		nil,
+		nil,
+	)
+
+	r, err := e.sqlEngine.QueryPreparedStmt(ctx, sqlTx, queryStmt, nil)
```

Comment:
it may be better to directly use sql delete statement

Reply:
resolved